src/overlays/actors/ovl_En_Insect/z_en_insect.c

```diff
@@ -21,55 +44,748 @@ const ActorInit En_Insect_InitVars = {
+}
+
+/**
+ * Find the nearest soft dirt patch within 6400 units in the xz plane and the current room
+ * Returns 1 if one was found, 0 otherwise
+ */
+s32 EnInsect_FindNearestSoil(EnInsect* this, GlobalContext* globalCtx) {
+    Actor* currentActor;
+    f32 currentDistance;
+    f32 bestDistance;
+    s32 ret;
+
+    ret = 0;
+    currentActor = globalCtx->actorCtx.actorList[ACTORTYPE_ITEMACTION].first;
+    bestDistance = 6400.0f;
+    this->soilActor = NULL;
+
+    while (currentActor != NULL) {
+        if (currentActor->id == ACTOR_OBJ_MAKEKINSUTA) {
+            currentDistance = func_800CB594(this->actor.posRot.pos.x, this->actor.posRot.pos.z,
+                                            currentActor->posRot.pos.x, currentActor->posRot.pos.z);
+
+            if (currentDistance < bestDistance && currentActor->room == this->actor.room) {
+                ret = 1;
+                bestDistance = currentDistance;
+                this->soilActor = (ObjMakekinsuta*)currentActor;
+            }
+        }
+        currentActor = currentActor->next;
+    }
+    return ret;
+}
+
+void func_80A7C058(EnInsect* this) {
+    if (this->unk_31E > 0) {
+        this->unk_31E -= 1;
+        return;
+    }
+
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_MUSI_WALK);
+
+    this->unk_31E = 3.0f / CLAMP_MIN(this->skelAnime.animPlaybackSpeed, 0.1f);
+    if (this->unk_31E < 2) {
+        this->unk_31E = 2;
+    }
+}
+
+void EnInsect_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnInsect* this = THIS;
+    f32 randomResult;
+    s16 temp_s2;
+    s32 count;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    func_80A7BE20(this);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/func_80A7BE6C.s")
+    temp_s2 = this->actor.params & 3;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/func_80A7BF58.s")
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_04035590, &D_040341FC, this->limbDrawTable,
+                   this->transitionDrawTable, 24);
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &sColliderInit, &this->colliderItem);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/func_80A7BFA0.s")
+    this->actor.colChkInfo.mass = 30;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/func_80A7C058.s")
+    if (this->unk_314 & 1) {
+        this->actor.gravity = -0.2f;
+        this->actor.minVelocityY = -2.0f;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/EnInsect_Init.s")
+    if (this->unk_314 & 4) {
+        this->unk_31C = Math_Rand_S16Offset(200, 40);
+        this->actor.flags |= 0x10;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/EnInsect_Destroy.s")
+    if (temp_s2 == 2 || temp_s2 == 3) {
+        if (EnInsect_FindNearestSoil(this, globalCtx) != 0) {
```

Comment:
since this function returns a boolean I think "EnInsect_FoundNearbySoil" might be good, and you would get rid of the != 0 in the conditional